src/front/spv/mod.rs

```diff
@@ -341,6 +328,8 @@ struct LookupVariable {
 struct LookupExpression {
     handle: Handle<crate::Expression>,
     type_id: spirv::Word,
+    /// The block id this expression belongs to
+    block_id: spirv::Word,
```

Comment:
I don't know if we could make it `NonZeroU32` since theoretically you could define a label with id 0? I don't know if spirv defines any ordering between ids

Reply:
> theoretically you could define a label with id 0?

In SPIR-V, id `0` is illegal: in the section `2.3 Physical Layout of a SPIR-V Module and Instruction`, `Table 1. First Words of Physical Layout`, section on `Bound`, it states:

> Bound; where all \<id>s in this module are guaranteed to satisfy
> 0 < id < Bound
> Bound should be small, smaller is better, with all \<id> in a module being densely packed and near 0.

note the `<` - it's `0 < id`, not `0 <= id`.

Reply:
Ah thank you, didn't notice that spirv defined that all ids must be larger than 0